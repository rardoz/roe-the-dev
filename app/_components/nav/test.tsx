import Nav from '.'
import { render } from '@testing-library/react'
import NavLinks from './nav-links'
import { mockIntersectionObserver } from 'jsdom-testing-mocks'
mockIntersectionObserver()

describe('Nav', () => {
  it('should render', () => {
    expect(
      render(
        <Nav messages={{ menuIconAltText: 'menu icon', logoAltText: 'logo' }}>
          <NavLinks>
            <a href="#">test</a>
          </NavLinks>
        </Nav>,
      ).container,
    ).toMatchSnapshot()
  })
})
