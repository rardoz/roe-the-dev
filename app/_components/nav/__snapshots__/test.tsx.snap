// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render 1`] = `
<div>
  <span
    class="nav-observer"
  />
  <nav
    class="bg-white z-20 border-gray-200 bg-transparent fixed top-0 w-full transition-color duration-500"
  >
    <div
      class="flex flex-wrap items-center justify-between mx-auto p-4 relative"
    >
      <a
        class="flex items-center space-x-3 rtl:space-x-reverse"
        href="/en-US"
        hreflang="en-US"
      >
        <img
          alt="logo"
          data-nimg="1"
          decoding="async"
          height="60"
          loading="eager"
          src="/logo.svg"
          style="color: transparent; transition: transform 0.25s ease-in-out;"
          width="60"
        />
      </a>
      <div
        class="flex-grow flex justify-end mx-3"
      >
        <span
          class="flex h-fit items-center gap-1 font-semibold bg-gray-200 text-gray-800 dark:bg-gray-400 dark:text-gray-900 group-hover:bg-gray-300 dark:group-hover:bg-gray-500 text-xs p-1.5 rounded-full"
          data-testid="flowbite-badge"
        >
          <span>
            <span>
              <a
                href="/en-US"
                hreflang="en-US"
              >
                <span
                  class="sr-only"
                >
                  en-US
                </span>
                ðŸ‡ºðŸ‡²
              </a>
            </span>
            <span>
               | 
              <a
                href="/es-US"
                hreflang="es-US"
              >
                <span
                  class="sr-only"
                >
                  es-US
                </span>
                ðŸ‡²ðŸ‡½
              </a>
            </span>
          </span>
        </span>
      </div>
      <button
        aria-expanded="false"
        class="text-gray-500 transition-all duration-500 inline-flex items-center justify-center p-2 w-10 h-10 text-sm  rounded-lg  hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
        type="button"
      >
        <span
          class="sr-only"
        >
          menu icon
        </span>
        <svg
          class="w-12 h-12"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
          />
        </svg>
      </button>
      <div
        class="hidden fixed top-0 left-0 w-screen h-screen bg-black bg-opacity-40"
      >
        <ul
          class="
              
              -right-56
          duration-300
          w-56
          absolute
          h-full
          flex
          flex-col
          font-medium
        bg-gray-50
        dark:bg-gray-800
        dark:border-gray-700
      "
        >
          <li
            class="flex items-center justify-end p-4"
          >
            <button
              aria-expanded="false"
              class="inline-flex items-center justify-center p-2 w-10 h-10 text-sm text-gray-500 rounded-lg hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
              type="button"
            >
              <span
                class="sr-only"
              >
                menu icon
              </span>
              <svg
                class="h-12 w-12"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 384 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M376.6 84.5c11.3-13.6 9.5-33.8-4.1-45.1s-33.8-9.5-45.1 4.1L192 206 56.6 43.5C45.3 29.9 25.1 28.1 11.5 39.4S-3.9 70.9 7.4 84.5L150.3 256 7.4 427.5c-11.3 13.6-9.5 33.8 4.1 45.1s33.8 9.5 45.1-4.1L192 306 327.4 468.5c11.3 13.6 31.5 15.4 45.1 4.1s15.4-31.5 4.1-45.1L233.7 256 376.6 84.5z"
                />
              </svg>
            </button>
          </li>
          <li>
            <a
              href="#"
            >
              test
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
